### DISCUSSION TITLE
"Your PC/Device needs to be repaired" upon first boot after using UTM "Create VM" wizard running Windows 11 arm64 on Apple Silicon

### DISCUSSION THREAD

**Discussion by brandonros** [utmapp/UTM#3595] [UTM Help (macOS)]
**Describe the issue**

"Your PC/Device needs to be repaired" upon first boot after using UTM "Create VM" wizard running Windows 11 arm64 on Apple Silicon

![image](https://user-images.githubusercontent.com/8949910/152700256-3340f145-69ff-40d9-b7ca-e17d5b2ebce0.png)

Followed the UTM wizard to create an arm64 Windows 11 VM. Downloaded `Windows11_InsiderPreview_Client_ARM64_en-us_22533.VHDX` as per the wizard. Immediately on first boot, doesn't work (Windows level error).

**Configuration**
* UTM Version: v3.0.4
* OS Version: Mac OS x 12.1
* Intel or Apple Silicon: Apple Silicon

**Crash log**

No.

**Debug log**

Not applicable.

**Upload VM**

```xml
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
<plist version="1.0">
<dict>
	<key>ConfigurationVersion</key>
	<integer>2</integer>
	<key>Debug</key>
	<dict/>
	<key>Display</key>
	<dict>
		<key>ConsoleFont</key>
		<string>Menlo</string>
		<key>ConsoleFontSize</key>
		<integer>12</integer>
		<key>ConsoleOnly</key>
		<false/>
		<key>ConsoleTheme</key>
		<string>Default</string>
		<key>DisplayCard</key>
		<string>virtio-ramfb</string>
		<key>DisplayDownscaler</key>
		<string>linear</string>
		<key>DisplayUpscaler</key>
		<string>nearest</string>
	</dict>
	<key>Drives</key>
	<array>
		<dict>
			<key>DriveName</key>
			<string>drive0</string>
			<key>ImagePath</key>
			<string>Windows11_InsiderPreview_Client_ARM64_en-us_22533.qcow2</string>
			<key>ImageType</key>
			<string>disk</string>
			<key>InterfaceType</key>
			<string>nvme</string>
		</dict>
		<dict>
			<key>DriveName</key>
			<string>cdrom0</string>
			<key>ImageType</key>
			<string>cd</string>
			<key>InterfaceType</key>
			<string>usb</string>
			<key>Removable</key>
			<true/>
		</dict>
	</array>
	<key>Info</key>
	<dict>
		<key>Icon</key>
		<string>windows</string>
	</dict>
	<key>Input</key>
	<dict/>
	<key>Networking</key>
	<dict>
		<key>NetworkCard</key>
		<string>virtio-net-pci</string>
		<key>NetworkCardMAC</key>
		<string>DE:E0:91:81:99:7B</string>
		<key>NetworkMode</key>
		<string>shared</string>
	</dict>
	<key>Printing</key>
	<dict/>
	<key>Sharing</key>
	<dict>
		<key>ClipboardSharing</key>
		<true/>
		<key>DirectoryReadOnly</key>
		<false/>
		<key>DirectorySharing</key>
		<true/>
		<key>Usb3Support</key>
		<false/>
		<key>UsbRedirectMax</key>
		<integer>3</integer>
	</dict>
	<key>Sound</key>
	<dict>
		<key>SoundCard</key>
		<string>intel-hda</string>
		<key>SoundEnabled</key>
		<true/>
	</dict>
	<key>System</key>
	<dict>
		<key>Architecture</key>
		<string>aarch64</string>
		<key>BootDevice</key>
		<string></string>
		<key>BootUefi</key>
		<true/>
		<key>CPU</key>
		<string>default</string>
		<key>CPUCount</key>
		<integer>4</integer>
		<key>MachineProperties</key>
		<string>highmem=off</string>
		<key>Memory</key>
		<integer>4096</integer>
		<key>RngEnabled</key>
		<true/>
		<key>SystemUUID</key>
		<string>4438A074-CE98-4472-891A-2480D7941D3A</string>
		<key>Target</key>
		<string>virt</string>
		<key>UseHypervisor</key>
		<true/>
	</dict>
</dict>
</plist>
```

**Comment by brandonros** [comment]
![image](https://user-images.githubusercontent.com/8949910/152700312-396cf8a3-b607-45b9-9f62-01b8caccb8d8.png)

```sh
qemu-system-aarch64
 -L /Applications/UTM.app/Contents/Resources/qemu
 -S
 -qmp tcp:127.0.0.1:4444,server,nowait
 -nodefaults
 -vga none
 -spice "unix=on,addr=/Users/brandonros/Library/Group Containers/WDNLXAD4W8.com.utmapp.UTM/4438A074-CE98-4472-891A-2480D7941D3A.spice,disable-ticketing=on,image-compression=off,playback-compression=off,streaming-video=off,gl=off"
 -device virtio-ramfb
 -device virtio-rng-pci
 -cpu host
 -smp cpus=4,sockets=1,cores=4,threads=1
 -machine virt,highmem=off
 -accel hvf
 -accel tcg,tb-size=1024
 -drive if=pflash,format=raw,unit=0,file=/Applications/UTM.app/Contents/Resources/qemu/edk2-aarch64-code.fd,readonly=on
 -drive if=pflash,format=raw,unit=1,file=/Users/brandonros/Library/Containers/com.utmapp.UTM/Data/Documents/Windows.utm/Images/efi_vars.fd
 -boot menu=on
 -m 4096
 -device intel-hda
 -device hda-duplex
 -name Windows
 -device nec-usb-xhci,id=usb-bus
 -device usb-tablet,bus=usb-bus.0
 -device usb-mouse,bus=usb-bus.0
 -device usb-kbd,bus=usb-bus.0
 -device ich9-usb-ehci1,id=usb-controller-0
 -device ich9-usb-uhci1,masterbus=usb-controller-0.0,firstport=0,multifunction=on
 -device ich9-usb-uhci2,masterbus=usb-controller-0.0,firstport=2,multifunction=on
 -device ich9-usb-uhci3,masterbus=usb-controller-0.0,firstport=4,multifunction=on
 -chardev spicevmc,name=usbredir,id=usbredirchardev0
 -device usb-redir,chardev=usbredirchardev0,id=usbredirdev0,bus=usb-controller-0.0
 -chardev spicevmc,name=usbredir,id=usbredirchardev1
 -device usb-redir,chardev=usbredirchardev1,id=usbredirdev1,bus=usb-controller-0.0
 -chardev spicevmc,name=usbredir,id=usbredirchardev2
 -device usb-redir,chardev=usbredirchardev2,id=usbredirdev2,bus=usb-controller-0.0
 -device nvme,drive=drive0,serial=drive0,bootindex=0
 -drive if=none,media=disk,id=drive0,file=/Users/brandonros/Library/Containers/com.utmapp.UTM/Data/Documents/Windows.utm/Images/Windows11_InsiderPreview_Client_ARM64_en-us_22533.qcow2,cache=writethrough
 -device usb-storage,drive=cdrom0,removable=true,bootindex=1,bus=usb-bus.0
 -drive if=none,media=cdrom,id=cdrom0
 -device virtio-net-pci,mac=DE:E0:91:81:99:7B,netdev=net0
 -netdev vmnet-macos,mode=shared,id=net0
 -device virtio-serial
 -device virtserialport,chardev=vdagent,name=com.redhat.spice.0
 -chardev spicevmc,id=vdagent,debug=0,name=vdagent
 -device virtserialport,chardev=charchannel1,id=channel1,name=org.spice-space.webdav.0
 -chardev spiceport,name=org.spice-space.webdav.0,id=charchannel1
 -uuid 4438A074-CE98-4472-891A-2480D7941D3A
 -rtc base=localtime
```

I didn't make any changes to the default settings from the wizard.

**Comment by brandonros** [comment]
For anybody else who may be using search to land on this issue:

File: \Windows\system32\winload.efi

Error code: 0xc000000f

**Reply by nimbus2300** [reply]
For me, this error was simply caused by running out of disk space when I made the virtual machine. I freed up space, started from scratch, and the error went away.

**Comment by geo404** [comment]
I had the same issue. The step from this video just helped me:
https://www.youtube.com/watch?v=FWT5RGt26Ag

> Homebrew Conversion (VHDX to QCOW2)
> 1. Install Homebrew using the install command on website: https://brew.sh/
> 2. install QEMU: brew install qemu
> 3. Run: qemu-img convert -p -O qcow2 (path) (path rename)

Don’t know how to attach configuration, screenshots instead
![utm1](https://user-images.githubusercontent.com/99163801/152735292-0a223224-67d3-4c57-b1b8-5b4b59247034.jpg)
![utm2](https://user-images.githubusercontent.com/99163801/152735301-b0a5de00-08cd-415c-9cd4-49f4fe8848c5.jpg)
![utm3](https://user-images.githubusercontent.com/99163801/152735307-32d67d70-59ca-43f0-9170-7583ec7c6f61.jpg)
.

**Comment by brandonros** [comment]
![image](https://user-images.githubusercontent.com/8949910/152813278-ca5b1eae-edd4-40f7-b7ed-c2ea8c4f1a5f.png)


I don't think that's it. It looks like the wizard does this automatically. Unless you're saying the .qcow2 file made by the UTM app/wizard is bad and the one from `qemu-img convert` works?

I'm not trying boot the VHDX from what I can tell.

**Comment by osy** [comment]
The latest preview may be broken on QEMU. Can you try the UUP method and preferably on a lower version?

**Comment by IDTitanium** [comment]
I have been battling the same issue since yesterday. I have tried to boot from VDHX and QCOW2, it still shows the same error. "Your PC/Device needs to be repaired."

**Comment by loscoto** [comment]
I had the same issue with UTM 3.1.5 and Windows11_InsiderPreview_Client_ARM64_en-us_22598, but it happened when I had what I thought was enough disk space available of ~17Gb. I freed up ~22Gb in total, and the first run succeeded. I assume the image conversion tries to perform its work, but runs out of available space and the silently fails (I didn't check the logs).

**Comment by garthk** [comment]
I'm not sure this is specific to Windows ARM64, Windows 11, or running out of disk space. I'm getting the same behaviour, predictably, with a VM configured for emulation and the x86_64 ISO for Windows 10–11. I've plenty of free disk space, and have confirmed the HD is in `.qcow2`.

Update: per #5638 I'll avoid installing SPICE on my next attempt.
